independent-projects/tools/analytics-common/src/main/java/io/quarkus/analytics/AnalyticsService.java

```diff
@@ -0,0 +1,260 @@
+        if (INSTANCE == null) {
+            INSTANCE = new AnalyticsService(fileLocations, log);
+        }
+        return INSTANCE;
+    }
+
+    public static boolean isInstantiated() {
+        return INSTANCE != null;
+    }
+
+    public static AnalyticsService getInstance() {
+        return INSTANCE;
+    }
+
+    final private RestClient restClient;
+    final private ConfigService config;
+    final private AnonymousUserId userId;
+    final private MessageWriter log;
+    final FileLocations fileLocations;
+
+    // Singleton
+    private AnalyticsService(final FileLocations fileLocations, MessageWriter log) {
+        this.fileLocations = fileLocations;
+        this.log = log;
+        this.postFutures = new ConcurrentLinkedQueue<>();
+        this.restClient = new RestClient(log);
+        this.userId = AnonymousUserId.getInstance(fileLocations, log);
+        this.config = new ConfigService(this.restClient, this.userId, fileLocations, log);
+    }
+
+    public void buildAnalyticsUserInput(Function<String, Boolean> analyticsEnabledSupplier) {
+        this.config.userAcceptance(analyticsEnabledSupplier);
+    }
+
+    public void sendAnalytics(final TrackEventType trackEventType,
+            ApplicationModel applicationModel,
+            final Map<String, Object> buildInfo,
+            final File localBuildDir) {
+
+        if (this.config.isActive() &&
+                this.config.isArtifactActive(
+                        applicationModel.getApplicationModule().getId().getGroupId(),
+                        getQuarkusVersion(applicationModel))) {
+
+            final Map<String, Object> context = createContextMap(applicationModel, buildInfo);
+            sendIdentity(context);
+            Track trackEvent = Track.builder()
+                    .userId(userId.getUuid())
+                    .context(context)
+                    .event(trackEventType)
+                    .properties(TrackProperties.builder()
+                            .appExtensions(createExtensionsPropertyValue(applicationModel))
+                            .build())
+                    .timestamp(Instant.now())
+                    .build();
+            postFutures.add(this.restClient.postTrack(trackEvent));
+            try {
+                FileUtils.overwrite(trackEvent,
+                        Path.of(localBuildDir.getAbsolutePath(), fileLocations.lastTrackFileName()));
+            } catch (IOException e) {
+                log.warn("Failed to write the last analytics file. " + e.getMessage());
+            }
+        }
+    }
+
+    public void cleanup() {
+        try {
+            // complete all. Normally, the queue should have only 1 element.
+            CompletableFuture.allOf(postFutures.toArray(new CompletableFuture[0])).get(TTL, TimeUnit.MILLISECONDS);
+        } catch (Exception e) {
+            log.warn("No analytics sent. " + e.getClass().getName() + ": " +
```

Comment:
in maven this just shows up as:

```
[INFO] [stdout] [warn] No analytics sent. java.util.concurrent.ExecutionException: java.net.http.HttpTimeoutException: request timed out
```

few issues:
can't see that this is related to the quarkus build analytics.
even with -X you don't get exception nor info on what host having the problem.

Reply:
Will improve the debug and warn logging.
The timeout is hardcoded... I can make it configurable to a larger value as well.

Reply:
Just a slight increase... It seems to be happening to other ppl as well.